src/librustc_errors/lib.rs

```diff
@@ -786,7 +786,10 @@ impl HandlerInner {
                 .emitted_diagnostic_codes
                 .iter()
                 .filter_map(|x| match &x {
-                    DiagnosticId::Error(s) if registry.find_description(s).is_some() => {
+                    DiagnosticId::Error(s)
+                        if registry.try_find_description(s).is_ok()
+                            && registry.try_find_description(s).unwrap().is_some() =>
```

Comment:
I think we can refactor this to move the try_find_description calls into the right hand side of the match arm, something like this:

```rust
DiagnosticId::Error(s) => {
    if let Ok(Some(explanation)) = registry.try_find_description(s) {
        Some(s.clone())
    } else {
        None
    }
}
```